packages/pymedphys_deliverydata/src/pymedphys_deliverydata/base/core.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (C) 2018 Cancer Care Associates
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version (the "AGPL-3.0+").
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License and the additional terms for more
+# details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+# ADDITIONAL TERMS are also included as allowed by Section 7 of the GNU
+# Affero General Public License. These additional terms are Sections 1, 5,
+# 6, 7, 8, and 9 from the Apache License, Version 2.0 (the "Apache-2.0")
+# where all references to the definition "License" are instead defined to
+# mean the AGPL-3.0+.
+
+# You should have received a copy of the Apache-2.0 along with this
+# program. If not, see <http://www.apache.org/licenses/LICENSE-2.0>.
+
+"""Working with delivery data from either logfiles or Mosaiq.
+"""
+
+from collections import namedtuple
+
+# Make the creation of DeliveryData look a bit like this...
+# Or a bit like the dataclass ...
+DeliveryDataBase = namedtuple(
+    'DeliveryData',
```

Comment:
The DeliveryData object is still a namedtuple at its core.

DeliveryDataBase is used throughout the pymedphys_deliverydata so as to avoid an internal circular dependency. Then, at a top level of pymedphys_deliverydata the functions within the module are attached to the primary DeliveryData object which is then used by every other package.

Reply:
As you can see, DeliveryDataBase is only used internally:

![image](https://user-images.githubusercontent.com/6559099/58230214-0893e080-7d77-11e9-9416-b59647c22ec2.png)

...in fact, maybe I should prepend it with _

Reply:
I'm not sure I see the circular dependency issue here. Can you explain what code specifically depends on this, and why it would in turn depend on that code if not separated as a "Base" object? Is it an artefact of preferring functions over methods? If so... why not stick to methods? :p